binzx/otomi

```diff
@@ -41,37 +41,73 @@ silent() {
     "$@" &>/dev/null
   fi
 }
+
+docker_tag_exists() {
+  curl -sflL https://index.docker.io/v1/repositories/$1/tags/$2 &>/dev/null
+}
+
 # shellcheck disable=SC2155
 readonly base_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
 
-if [[ $base_dir == *"otomi-core"* ]] || [[ $(pwd) == *"otomi-core"* ]]; then
+findUp() {
+  path=$1
+  while [[ "$path" != "" && ! -e "$path/$2" ]]; do
+    path=${path%/*}
+  done
+  echo "$path"
+}
+
+packageIsCore() {
+  path=$1
+  [ -z $path ] && return 1
+  packageName=$(grep name "$path/package.json" | sed 's/.*"name": "\(.*\)".*/\1/')
+  [[ $packageName == 'otomi-core' ]] && return 0 || return 1
+}
+
+packageJsonPathPwd=$(findUp $(pwd) 'package.json')
+packageJsonPathBaseDir=$(findUp $base_dir 'package.json')
+
+if packageIsCore $packageJsonPathPwd; then
   in_core=1
+  core_path=$packageJsonPathPwd
+elif packageIsCore $packageJsonPathBaseDir; then
+  in_core=1
+  core_path=$packageJsonPathBaseDir
 fi
```

Comment:
I am not sure if this is doing the same, but I would like to check that package.json exists in these locations in this order:
1. `./package.json`
2. `ENV_DIR/package.json`

Reply:
It does that. 

But for point 1 it finds the first package.json from the current directory up. So

./package.json
../package.json
../../package.json
Etc...

It picks the one with the least dots. And checks if it is otomi-core